app/src/main/java/nerd/tuxmobil/fahrplan/congress/alarms/AlarmReceiver.kt

```diff
@@ -0,0 +1,211 @@
+
+import android.net.Uri
+import android.app.PendingIntent
+import android.content.Intent
+import android.content.Context
+import android.content.BroadcastReceiver
+import androidx.annotation.VisibleForTesting
+
+class AlarmReceiver : BroadcastReceiver() {
+
+    companion object {
+        const val ALARM_UPDATE = "nerd.tuxmobil.fahrplan.congress.ALARM_UPDATE"
+        const val ALARM_SESSION = "nerd.tuxmobil.fahrplan.congress.ALARM_SESSION"
+        const val ALARM_DELETE = "de.machtnix.fahrplan.ALARM"
+        const val ALARM_DISMISSED = "nerd.tuxmobil.fahrplan.congress.ALARM_DISMISSED"
+        private const val LOG_TAG = "AlarmReceiver"
+        private const val BUNDLE_KEY_NOTIFICATION_ID = "BUNDLE_KEY_NOTIFICATION_ID"
+        private const val INVALID_NOTIFICATION_ID = -1
+        private const val DEFAULT_REQUEST_CODE = 0
+
+        /**
+         * Returns a unique [Intent] to delete the data associated with the
+         * given session alarm `notificationId`.
+         * The `Intent` is composed with a fake data URI to comply with the uniqueness rules
+         * defined by [Intent.filterEquals].
+         */
+        private fun createDeleteNotificationIntent(context: Context, notificationId: Int) =
+            Intent(context, AlarmReceiver::class.java).apply {
+                action = ALARM_DISMISSED
+                putExtra(BUNDLE_KEY_NOTIFICATION_ID, notificationId)
+                data = Uri.parse("fake://$notificationId")
+            }
+
+        private fun getAddAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long
+        ) = getAlarmIntent(context, sessionId, day, title, startTime, ALARM_SESSION)
+
+        private fun getDeleteAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long
+        ) = getAlarmIntent(context, sessionId, day, title, startTime, ALARM_DELETE)
+
+        private fun getAlarmIntent(
+            context: Context,
+            sessionId: String,
+            day: Int,
+            title: String,
+            startTime: Long,
+            intentAction: String
+        ) = Intent(context, AlarmReceiver::class.java).apply {
+            putExtra(BundleKeys.ALARM_SESSION_ID, sessionId)
+            putExtra(BundleKeys.ALARM_DAY, day)
+            putExtra(BundleKeys.ALARM_TITLE, title)
+            putExtra(BundleKeys.ALARM_START_TIME, startTime)
+            action = intentAction
+            data = Uri.parse("alarm://$sessionId")
+        }
+    }
+
+    private val logging = Logging.get()
+
+    override fun onReceive(context: Context, intent: Intent) {
+        logging.d(LOG_TAG, "Received alarm = " + intent.action + ".")
+
+        when (intent.action){
+            ALARM_DISMISSED -> onSessionAlarmNotificationDismissed(intent)
+            ALARM_UPDATE -> start(context)
```

Comment:
Please use the full qualified path `UpdateService.start(context)` so one can see what is started here.